server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -110,10 +122,20 @@
     private final Map<TcpChannel, ChannelPendingTaskTracker> channelPendingTaskTrackers = ConcurrentCollections.newConcurrentMap();
     private final SetOnce<TaskCancellationService> cancellationService = new SetOnce<>();
 
+    private final boolean taskResourceConsumersEnabled;
+    /** Consumers that are notified of the resource stats */
+    private final List<Consumer<Task>> taskResourceConsumer;
+
     public TaskManager(Settings settings, ThreadPool threadPool, Set<String> taskHeaders) {
         this.threadPool = threadPool;
         this.taskHeaders = new ArrayList<>(taskHeaders);
         this.maxHeaderSize = SETTING_HTTP_MAX_HEADER_SIZE.get(settings);
+        this.taskResourceConsumersEnabled = TASK_RESOURCE_CONSUMERS_SETTING.get(settings);
+        this.taskResourceConsumer = new ArrayList<Consumer<Task>>() {
+            {
+                add(new TopNSearchTasksLogger(settings));
+            }
+        };
```

Comment:
Nitpick - gate this with a check on `taskResourceConsumersEnabled` ?

Reply:
Adding TASK_RESOURCE_CONSUMER_SETTING to clusterSetting to enable/disable as required. Also adding a check to see if enabled before publishing to consumers. Do you still think we need to include a gate check here?

Reply:
As we continue to add more consumers what might help is a control knob per consumer instead of a blanket level knob

Reply:
@Bukhtawar When we have more consumers, it makes sense to have control knob per consumer! I think blanket level will also be helpful to control all consumers using one knob (like parent level). Going forward we can add the control knobs per consumer. Any thoughts?